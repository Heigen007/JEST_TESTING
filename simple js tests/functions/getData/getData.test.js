const getData = require('./getData.js');
const axios = require('axios');

jest.mock('axios');

describe('getData', () => {
    var response = {
        data: [
            {
                "userId": 1,
                "id": 1,
                "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
                "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
            },
                {
                "userId": 1,
                "id": 2,
                "title": "qui est esse",
                "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
            }
        ]
    }
    axios.get.mockReturnValue(response)

    test("Should return value", async () => {
        var res = await getData()
        expect(res).not.toBe(false)
    })

    test("response is an array", async () => {
        var res = await getData()
        expect(res).toBeInstanceOf(Array)
    })

    test("function works", async () => {
        var res = await getData()
        expect(res.length).toBe(2)
        expect(res).toMatchSnapshot()
    })
})
