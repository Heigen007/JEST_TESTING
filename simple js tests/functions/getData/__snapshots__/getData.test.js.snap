// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData function works 1`] = `
Array [
  "1",
  "2",
]
`;